#VRML_SIM R2019a utf8
WorldInfo {
  FPS 30
}
Viewpoint {
  orientation 0.35883700320980194 0.9089532641721003 0.21222622052492135 5.129494241773969
  position -1.3984486302154884 1.1316012845893137 -0.9587102928622189
}
Background {
  skyColor [
    0.8274509803921568 0.8431372549019608 0.8117647058823529
  ]
}
DEF robot Robot {
  translation 0.11999999805902845 0.23733518540619783 -1.9999999997429516
  rotation 0.13245558889851905 0.0003959832386026674 0.9911888620070451 8.117122753331102e-09
  children [
    DEF Torso Solid {
      translation 0 0.03 -0.1
      children [
        DEF IMU_Torso InertialUnit {
          translation 0.2599999999999999 0.08 0.11999999999999998
          name "IMU_Torso"
        }
        DEF FR_Shoulder_Hip Hinge2Joint {
          jointParameters HingeJointParameters {
            position -4.0354430450721357e-07
            anchor 0.395 0.025 0.2
            minStop -0.5
            maxStop 0.5
          }
          jointParameters2 JointParameters {
            position 7.375775934760641e-14
            minStop -3.14
            maxStop 3.14
          }
          device [
            DEF FR_Motor_Shoulder_Abduction RotationalMotor {
              name "rotational motor1"
              acceleration 1.5
              maxVelocity 3
              minPosition -0.5
              maxPosition 0.5
            }
            PositionSensor {
              name "pos_sens1"
            }
          ]
          device2 [
            DEF FR_Motor_Shoulder_Rotation RotationalMotor {
              name "rotational motor2"
              acceleration 1.5
              maxVelocity 3
              minPosition -3.14
              maxPosition 3.14
            }
            PositionSensor {
              name "pos_sens2"
            }
          ]
          endPoint Solid {
            translation 0.395 0.02500000504430381 0.212499999999999
            rotation -4.7936050553485126e-14 0.9999999999999797 -2.0177215225360517e-07 3.141592644
            children [
              DEF upperlegbodyconnector Shape {
                appearance PBRAppearance {
                  baseColor 0.04 1 0.68
                }
                geometry Box {
                  size 0.05 0.05 0.025
                }
              }
              Solid {
                translation 0 -0.065 -0.0275
                children [
                  DEF upperLeg Shape {
                    appearance PBRAppearance {
                    }
                    geometry Box {
                      size 0.05 0.18 0.03
                    }
                  }
                  Transform {
                    translation 0 -0.11 0.026
                    children [
                      SliderJoint {
                        jointParameters JointParameters {
                          position 0.010033844294535136
                          axis 0 1 0
                          minStop -0.01
                          maxStop 0.085
                        }
                        device [
                          DEF FR_Motor_Knee LinearMotor {
                            name "linear motor1"
                            acceleration 0.75
                            maxVelocity 0.2
                            minPosition -0.01
                            maxPosition 0.085
                            maxForce 100
                          }
                          PositionSensor {
                            name "pos_sens1_lin"
                          }
                        ]
                        endPoint Solid {
                          translation 0 0.010033844294535136 0
                          children [
                            SliderJoint {
                              jointParameters JointParameters {
                                position 0.01177197431146824
                                axis 0 1 0
                                maxStop 0.04
                                springConstant 3000
                              }
                              endPoint Solid {
                                translation 0 -0.07822802568853175 0
                                children [
                                  DEF FR_Force TouchSensor {
                                    translation 0 -0.011 0
                                    rotation 1 0 0 1.5707963220000003
                                    name "FR_Force"
                                    boundingObject Sphere {
                                      radius 0.015
                                    }
                                    physics Physics {
                                    }
                                    type "force-3d"
                                  }
                                  DEF FR_Foot Shape {
                                    appearance PBRAppearance {
                                      baseColor 1 0.2 0.1
                                    }
                                    geometry Capsule {
                                      height 0.03
                                      radius 0.01
                                    }
                                  }
                                ]
                                boundingObject USE FR_Foot
                                physics Physics {
                                  density -1
                                  mass 0.1
                                }
                              }
                            }
                            DEF lowerandupperlegconnector Shape {
                              appearance PBRAppearance {
                                baseColor 0.5 0.9 0.1
                              }
                              geometry Box {
                                size 0.025 0.06 0.02
                              }
                            }
                          ]
                          boundingObject USE lowerandupperlegconnector
                          physics Physics {
                            mass 0.15
                          }
                        }
                      }
                    ]
                  }
                ]
                boundingObject USE upperLeg
                physics Physics {
                  mass 3
                }
              }
            ]
            name "solid(3)"
            boundingObject USE upperlegbodyconnector
            physics Physics {
              mass 0.03
            }
          }
        }
        DEF BR_Shoulder_Hip Hinge2Joint {
          jointParameters HingeJointParameters {
            position -4.0354436621567405e-07
            anchor 0.12 0.025 0.2
            minStop -0.5
            maxStop 0.5
          }
          jointParameters2 JointParameters {
            position 9.20819810197201e-14
            minStop -3.14
            maxStop 3.14
          }
          device [
            DEF BR_Motor_Shoulder_Abduction RotationalMotor {
              name "rotational motor3"
              acceleration 1.5
              maxVelocity 3
              minPosition -0.5
              maxPosition 0.5
            }
            PositionSensor {
              name "pos_sens3"
            }
          ]
          device2 [
            DEF BR_Motor_Shoulder_Rotation RotationalMotor {
              name "rotational motor4"
              acceleration 1.5
              maxVelocity 3
              minPosition -3.14
              maxPosition 3.14
            }
            PositionSensor {
              name "pos_sens4"
            }
          ]
          endPoint Solid {
            translation 0.12000000057538757 0.02500000504430458 0.212499999999999
            rotation -6.623343088793944e-14 0.9999999999999797 -2.0177218310783533e-07 3.141592644
            children [
              DEF upperlegbodyconnector Shape {
                appearance PBRAppearance {
                  baseColor 0.04 1 0.68
                }
                geometry DEF boxer1 Box {
                  size 0.05 0.05 0.025
                }
              }
              Solid {
                translation 0 -0.065 -0.0275
                children [
                  DEF upperLeg Shape {
                    appearance PBRAppearance {
                    }
                    geometry Box {
                      size 0.05 0.18 0.03
                    }
                  }
                  Transform {
                    translation 0 -0.11 0.026
                    children [
                      SliderJoint {
                        jointParameters JointParameters {
                          position 0.010033844298977055
                          axis 0 1 0
                          minStop -0.01
                          maxStop 0.085
                        }
                        device [
                          DEF BR_Motor_Knee LinearMotor {
                            name "linear motor4"
                            acceleration 0.75
                            maxVelocity 0.2
                            minPosition -0.01
                            maxPosition 0.085
                            maxForce 100
                          }
                          PositionSensor {
                            name "pos_sens4_lin"
                          }
                        ]
                        endPoint Solid {
                          translation 0 0.010033844298977055 0
                          children [
                            SliderJoint {
                              jointParameters JointParameters {
                                position 0.011771975818023259
                                axis 0 1 0
                                maxStop 0.04
                                springConstant 3000
                              }
                              endPoint Solid {
                                translation 0 -0.07822802418197673 0
                                children [
                                  DEF BR_Force TouchSensor {
                                    translation 0 -0.011 0
                                    rotation 1 0 0 1.5707963220000003
                                    name "BR_Force"
                                    boundingObject Sphere {
                                      radius 0.015
                                    }
                                    physics Physics {
                                    }
                                    type "force-3d"
                                  }
                                  DEF BR_Foot Shape {
                                    appearance PBRAppearance {
                                      baseColor 1 0.2 0.1
                                    }
                                    geometry Capsule {
                                      height 0.03
                                      radius 0.01
                                    }
                                  }
                                ]
                                boundingObject USE BR_Foot
                                physics Physics {
                                  density -1
                                  mass 0.1
                                }
                              }
                            }
                            DEF lowerandupperlegconnector Shape {
                              appearance PBRAppearance {
                                baseColor 0.5 0.9 0.1
                              }
                              geometry Box {
                                size 0.025 0.06 0.02
                              }
                            }
                          ]
                          boundingObject USE lowerandupperlegconnector
                          physics Physics {
                            density -1
                            mass 0.15
                          }
                        }
                      }
                    ]
                  }
                ]
                boundingObject USE upperLeg
                physics Physics {
                  density -1
                  mass 3
                }
              }
            ]
            name "solid(2)"
            boundingObject USE upperlegbodyconnector
            physics Physics {
              density -1
              mass 0.03
            }
          }
        }
        DEF FL_Shoulder_Hip Hinge2Joint {
          jointParameters HingeJointParameters {
            position 4.0354432411549764e-07
            anchor 0.395 0.025 0
            minStop -0.5
            maxStop 0.5
          }
          jointParameters2 JointParameters {
            position 7.239833469963069e-14
            minStop -3.14
            maxStop 3.14
          }
          device [
            DEF FL_Motor_Shoulder_Abduction RotationalMotor {
              name "rotational motor5"
              acceleration 1.5
              maxVelocity 3
              minPosition -0.5
              maxPosition 0.5
            }
            PositionSensor {
              name "pos_sens5"
            }
          ]
          device2 [
            DEF FL_Motor_Shoulder_Rotation RotationalMotor {
              name "rotational motor6"
              acceleration 1.5
              maxVelocity 3
              minPosition -3.14
              maxPosition 3.14
            }
            PositionSensor {
              name "pos_sens6"
            }
          ]
          endPoint Solid {
            translation 0.395 0.025000005044304052 -0.012499999999998988
            rotation 0.9999999999999838 -3.6199167349814756e-14 1.794061528639108e-07 4.031583445807906e-07
            children [
              DEF upperlegbodyconnector Shape {
                appearance PBRAppearance {
                  baseColor 0.04 1 0.68
                }
                geometry Box {
                  size 0.05 0.05 0.025
                }
              }
              Solid {
                translation 0 -0.065 -0.0275
                children [
                  DEF upperLeg Shape {
                    appearance PBRAppearance {
                    }
                    geometry Box {
                      size 0.05 0.18 0.03
                    }
                  }
                  Transform {
                    translation 0 -0.11 0.026
                    children [
                      SliderJoint {
                        jointParameters JointParameters {
                          position 0.010033844294014107
                          axis 0 1 0
                          minStop -0.01
                          maxStop 0.085
                        }
                        device [
                          DEF FL_Motor_Knee LinearMotor {
                            name "linear motor2"
                            acceleration 0.75
                            maxVelocity 0.2
                            minPosition -0.01
                            maxPosition 0.085
                            maxForce 100
                          }
                          PositionSensor {
                            name "pos_sens2_lin"
                          }
                        ]
                        endPoint Solid {
                          translation 0 0.010033844294014107 0
                          children [
                            SliderJoint {
                              jointParameters JointParameters {
                                position 0.01177197415192803
                                axis 0 1 0
                                maxStop 0.04
                                springConstant 3000
                              }
                              endPoint Solid {
                                translation 0 -0.07822802584807197 0
                                children [
                                  DEF FL_Force TouchSensor {
                                    translation 0 -0.011 0
                                    rotation 1 0 0 1.5707963220000003
                                    name "FL_Force"
                                    boundingObject Sphere {
                                      radius 0.015
                                    }
                                    physics Physics {
                                    }
                                    type "force-3d"
                                  }
                                  DEF FL_Foot Shape {
                                    appearance PBRAppearance {
                                      baseColor 1 0.2 0.1
                                    }
                                    geometry Capsule {
                                      height 0.03
                                      radius 0.01
                                    }
                                  }
                                ]
                                boundingObject USE FL_Foot
                                physics Physics {
                                  density -1
                                  mass 0.1
                                }
                              }
                            }
                            DEF lowerandupperlegconnector Shape {
                              appearance PBRAppearance {
                                baseColor 0.5 0.9 0.1
                              }
                              geometry Box {
                                size 0.025 0.06 0.02
                              }
                            }
                          ]
                          boundingObject USE lowerandupperlegconnector
                          physics Physics {
                            density -1
                            mass 0.15
                          }
                        }
                      }
                    ]
                  }
                ]
                boundingObject USE upperLeg
                physics Physics {
                  density -1
                  mass 3
                }
              }
            ]
            name "solid(1)"
            boundingObject USE upperlegbodyconnector
            physics Physics {
              density -1
              mass 0.03
            }
          }
        }
        DEF Torso Shape {
          appearance PBRAppearance {
            baseColor 1 0.2 0.65
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                0 0 0
                0 0 0.2
                0 0.05 0
                0 0.05 0.2
                0 0.14 0.04
                0 0.14 0.16
                0.515 0 0
                0.515 0 0.2
                0.515 0.05 0
                0.515 0.05 0.2
                0.515 0.14 0.04
                0.515 0.14 0.16
              ]
            }
            coordIndex [
              0, 1, 2, -1, 2, 1, 3, -1, 2, 3, 4, -1, 4, 3, 5, -1, 1, 7, 3, -1, 3, 7, 9, -1, 3, 9, 5, -1, 5, 9, 11, -1, 4, 5, 11, -1, 4, 11, 10, -1, 4, 10, 8, -1, 2, 4, 8, -1, 0, 2, 6, -1, 2, 8, 6, -1, 0, 6, 1, -1, 1, 6, 7, -1, 6, 8, 7, -1, 7, 8, 9, -1, 8, 10, 9, -1, 10, 11, 9
            ]
          }
        }
        DEF BL_Shoulder_Hip Hinge2Joint {
          jointParameters HingeJointParameters {
            position 4.035443835749079e-07
            anchor 0.12 0.025 0
            minStop -0.5
            maxStop 0.5
          }
          jointParameters2 JointParameters {
            position 7.242720362580034e-14
            minStop -3.14
            maxStop 3.14
          }
          device [
            DEF BL_Motor_Shoulder_Abduction RotationalMotor {
              name "rotational motor7"
              acceleration 1.5
              maxVelocity 3
              minPosition -0.5
              maxPosition 0.5
            }
            PositionSensor {
              name "pos_sens7"
            }
          ]
          device2 [
            DEF BL_Motor_Shoulder_Rotation RotationalMotor {
              name "rotational motor8"
              acceleration 1.5
              maxVelocity 3
              minPosition -3.14
              maxPosition 3.14
            }
            PositionSensor {
              name "pos_sens8"
            }
          ]
          endPoint Solid {
            translation 0.12 0.025000005044304795 -0.012499999999998988
            rotation 0.9999999999999839 -3.621360181289959e-14 1.7947766484613683e-07 4.031583445807906e-07
            children [
              DEF upperlegbodyconnector Shape {
                appearance PBRAppearance {
                  baseColor 0.04 1 0.68
                }
                geometry Box {
                  size 0.05 0.05 0.025
                }
              }
              Solid {
                translation 0 -0.065 -0.0275
                children [
                  DEF upperLeg Shape {
                    appearance PBRAppearance {
                    }
                    geometry Box {
                      size 0.05 0.18 0.03
                    }
                  }
                  Transform {
                    translation 0 -0.11 0.026
                    children [
                      SliderJoint {
                        jointParameters JointParameters {
                          position 0.010033844298455318
                          axis 0 1 0
                          minStop -0.01
                          maxStop 0.085
                        }
                        device [
                          DEF BL_Motor_Knee LinearMotor {
                            name "linear motor3"
                            acceleration 0.75
                            maxVelocity 0.2
                            minPosition -0.01
                            maxPosition 0.085
                            maxForce 100
                          }
                          PositionSensor {
                            name "pos_sens3_lin"
                          }
                        ]
                        endPoint Solid {
                          translation 0 0.010033844298455318 0
                          children [
                            SliderJoint {
                              jointParameters JointParameters {
                                position 0.011771975658484268
                                axis 0 1 0
                                maxStop 0.04
                                springConstant 3000
                              }
                              endPoint Solid {
                                translation 0 -0.07822802434151573 0
                                children [
                                  DEF BL_Force TouchSensor {
                                    translation 0 -0.011 0
                                    rotation 1 0 0 1.5707963220000003
                                    name "BL_Force"
                                    boundingObject Sphere {
                                      radius 0.015
                                    }
                                    physics Physics {
                                    }
                                    type "force-3d"
                                  }
                                  DEF BL_Foot Shape {
                                    appearance PBRAppearance {
                                      baseColor 1 0.2 0.1
                                    }
                                    geometry Capsule {
                                      height 0.03
                                      radius 0.01
                                    }
                                  }
                                ]
                                boundingObject USE BL_Foot
                                physics Physics {
                                  density -1
                                  mass 0.1
                                }
                              }
                            }
                            DEF lowerandupperlegconnector Shape {
                              appearance PBRAppearance {
                                baseColor 0.5 0.9 0.1
                              }
                              geometry Box {
                                size 0.025 0.06 0.02
                              }
                            }
                          ]
                          boundingObject USE lowerandupperlegconnector
                          physics Physics {
                            density -1
                            mass 0.15
                          }
                        }
                      }
                    ]
                  }
                ]
                boundingObject USE upperLeg
                physics Physics {
                  density -1
                  mass 3
                }
              }
            ]
            boundingObject USE upperlegbodyconnector
            physics Physics {
              density -1
              mass 0.03
            }
          }
        }
      ]
      boundingObject USE Torso
      physics Physics {
        density -1
        mass 15
      }
    }
    TouchSensor {
      translation 0 0.09999999999999999 0
      name "collision_detector"
      boundingObject Box {
        size 0.55 0.15 0.205
      }
    }
  ]
  physics Physics {
  }
  controller "RL_controller"
  controllerArgs "None"
  supervisor TRUE
  battery [
    5000000, 5000000, 0
  ]
}
Floor {
}
